### PR TITLE
AA-129: Resume Course button

### PR DESCRIPTION
**Pull request by ciduarte**
- Exposed Resume Course fields from API
- Made the Start/Resume Course button functional
- Freebies: fixed headers and alignment of a link in the Dates Widget

### CODE REVIEW

**Comment by ciduarte** [comment]
**Outline Tab Before**
AA-129-Outline_Tab_Before

**Outline Tab After**
AA-129-Outline_Tab_After

File: src/course-home/data/__factories__/outlineTabData.factory.js [hunk 1]
```diff
@@ -20,4 +20,8 @@ Factory.define('outlineTabData')
     can_enroll: true,
     extra_text: 'Contact the administrator.',
   })
-  .attr('handouts_html', [], () => '<ul><li>Handout 1</li></ul>');
+  .attr('handouts_html', [], () => '<ul><li>Handout 1</li></ul>')
+  .attr('resume_course', ['host', 'courseId'], (host, courseId) => ({
+    hasVisitedCourse: false,
```
Comment (mikix) [inline src/course-home/data/__factories__/outlineTabData.factory.js]: In this mock, I think we use snake case, in the expectation that our camel casing code will later kick in.

File: src/course-home/outline-tab/OutlineTab.jsx [hunk 2]
```diff
@@ -37,11 +37,14 @@ function OutlineTab({ intl }) {
       courses,
       sections,
     },
+    resumeCourse,
   } = useModel('outline', courseId);
 
   const clientEnrollmentAlert = useEnrollmentAlert(courseId);
   useLogistrationAlert();
 
+  const { hasVisitedCourse } = resumeCourse;
```
Comment (mikix) [inline src/course-home/outline-tab/OutlineTab.jsx]: You *could* do that decomposition above with:
```
resumeCourse: {
   hasVisitedCourse,
}
```

But I guess you use the resumeCourse object for its url later? Which feels slightly inconsistent. You could also do:

```
resumeCourse: {
   hasVisitedCourse,
   url: resumeUrl,
}
```

To rename the url object, if you were worried about such a generic variable name (I can never remember whether the new name is on left or right of the colon, but something like that).

**Review by mikix — APPROVED** [review]
Beautiful

File: src/course-home/outline-tab/OutlineTab.jsx (hunk 2 above)
Comment (ciduarte) [reply src/course-home/outline-tab/OutlineTab.jsx]: Yess you read my mind. I was trying to do that last suggestion but couldn't figure out the syntax. Thank you!!

**Pull request merged**